apps/web/components/dashboard/bookmarks/DeleteBookmarkConfirmationDialog.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect } from "react";
+import { usePathname, useRouter } from "next/navigation";
+import { ActionButton } from "@/components/ui/action-button";
+import ActionConfirmingDialog from "@/components/ui/action-confirming-dialog";
+import { toast } from "@/components/ui/use-toast";
+import { useTranslation } from "@/lib/i18n/client";
+
+import { useDeleteBookmark } from "@hoarder/shared-react/hooks//bookmarks";
+import { ZBookmark } from "@hoarder/shared/types/bookmarks";
+
+export default function DeleteBookmarkConfirmationDialog({
+  bookmark,
+  children,
+  open,
+  setOpen,
+  onPendingChange,
+}: {
+  bookmark: ZBookmark;
+  children?: React.ReactNode;
+  open: boolean;
+  setOpen: (v: boolean) => void;
+  onPendingChange?: (isPending: boolean) => void;
+}) {
+  const { t } = useTranslation();
+  const currentPath = usePathname();
+  const router = useRouter();
+
+  const { mutate: deleteBoomark, isPending } = useDeleteBookmark({
+    onSuccess: () => {
+      toast({
+        description: t("toasts.bookmarks.deleted"),
+      });
+      setOpen(false);
+      if (currentPath.includes(bookmark.id)) {
+        router.push("/dashboard/bookmarks");
+      }
+    },
+    onError: () => {
+      toast({
+        variant: "destructive",
+        description: `Something went wrong`,
+      });
+    },
+  });
+
+  useEffect(() => {
+    onPendingChange?.(isPending);
+  }, [isPending, onPendingChange]);
+
+  return (
+    <ActionConfirmingDialog
+      open={open}
+      setOpen={setOpen}
+      title={`Delete ${bookmark.title ?? "Bookmark"}?`}
+      description={`Are you sure you want to delete this bookmark?`}
```

Comment:
Can you add this to the translation files in `apps/web/lib/i18n/locales/en.json`? So that people can also get a chance to translate it?

Reply:
Fixed in [add delete bookmark description to i18n](https://github.com/hoarder-app/hoarder/pull/787/commits/52924c63425199140a4fba762518a2a8c01d1fc4)